pkg/xen-tools/patches-4.15.0/13-qemu-Add-cpu-pin-and-cpumask-options.patch

```diff
@@ -0,0 +1,128 @@
+From 0e950484e1da0b0e2800c80137cc53fa495e59d2 Mon Sep 17 00:00:00 2001
+From: Nikolay Martyanov <ohmspectator@gmail.com>
+Date: Wed, 28 Sep 2022 15:47:11 +0200
+Subject: [PATCH 13/15] qemu: Add 'cpu-pin' and 'cpumask' options.
+
+Signed-off-by: Nikolay Martyanov <ohmspectator@gmail.com>
+---
+ tools/qemu-xen/hw/core/machine.c   | 38 ++++++++++++++++++++++++++++++
+ tools/qemu-xen/include/hw/boards.h |  2 ++
+ tools/qemu-xen/qemu-options.hx     |  8 +++++++
+ tools/qemu-xen/softmmu/vl.c        |  4 ++++
+ 4 files changed, 52 insertions(+)
+
+diff --git a/tools/qemu-xen/hw/core/machine.c b/tools/qemu-xen/hw/core/machine.c
+index 8d1a90c..2f9993e 100644
+--- a/tools/qemu-xen/hw/core/machine.c
++++ b/tools/qemu-xen/hw/core/machine.c
+@@ -372,6 +372,35 @@ static void machine_set_graphics(Object *obj, bool value, Error **errp)
+     ms->enable_graphics = value;
+ }
+ 
++static bool machine_get_cpu_pin(Object *obj, Error **errp)
++{
++    MachineState *ms = MACHINE(obj);
++
++    return ms->cpu_pin;
++}
++
++static void machine_set_cpu_pin(Object *obj, bool value, Error **errp)
++{
++    MachineState *ms = MACHINE(obj);
++
++    ms->cpu_pin = value;
++}
++
++static char *machine_get_cpumask(Object *obj, Error **errp)
++{
++    MachineState *ms = MACHINE(obj);
++
++    return g_strdup(ms->cpumask_str);
++}
++
++static void machine_set_cpumask(Object *obj, const char *value, Error **errp)
++{
++    MachineState *ms = MACHINE(obj);
++
++    g_free(ms->cpumask_str);
++    ms->cpumask_str = g_strdup(value);
++}
++
+ static char *machine_get_firmware(Object *obj, Error **errp)
+ {
+     MachineState *ms = MACHINE(obj);
+@@ -841,6 +870,15 @@ static void machine_class_init(ObjectClass *oc, void *data)
+     object_class_property_set_description(oc, "usb",
+         "Set on/off to enable/disable usb");
+ 
++    object_class_property_add_bool(oc, "cpu-pin",
++        machine_get_cpu_pin, machine_set_cpu_pin);
++    object_class_property_set_description(oc, "cpu-pin",
++        "Set on/off to enable/disable CPU pinning");
++
++    object_class_property_add_str(oc, "cpumask",
++        machine_get_cpumask, machine_set_cpumask);
++    object_class_property_set_description(oc, "cpumask", "CPU Mask");
```

Comment:
I think it would be useful to add an example to the description on how CPU Mask should look like, for example:

"CPU Mask, e.g. 00111001"

Reply:
Ok, I'll put the example in the `.hx` file. Taking a look at the code around, a short description is more proper here, I think.